cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -124,11 +124,14 @@ func CreateStaticPodFiles(manifestDir string, cfg *kubeadmapi.ClusterConfigurati
 		}
 
 		// writes the StaticPodSpec to disk
-		if err := staticpodutil.WriteStaticPodToDisk(componentName, manifestDir, spec); err != nil {
-			return errors.Wrapf(err, "failed to create static pod manifest file for %q", componentName)
+		if !staticpodutil.CheckStaticPodExists(componentName, manifestDir) {
```

Comment:
Please take a look at https://github.com/kubernetes/kubernetes/pull/80404#discussion_r305656699

Reply:
Agreed, a small refactoring of this to work like the check-or-create functions in the other phases makes sense.